src/responder/multiResponder.ts

```diff
@@ -370,35 +192,28 @@ export class MultiResponder extends StartStopService {
      * issued to ensure that all responses are made.
      * @param queueItems
      */
-    public async reEnqueueMissingItems(appointmentIds: string[]) {
+    public async reEnqueueMissingItems(appointmentIdsStillPending: string[]) {
         // a reorg may have occurred, if this is the case then we need to check whether
         // then some transactions that we had previously considered mined may no longer
         // be. We can find these transactions by comparing the current gas queue to the
         // transactions that we currently observe in pending. Transactions in pending
         // but not in the gas queue need to be added there.
-
-        const missingQueueItems: GasQueueItemRequest[] = [];
-        const queueItems: GasQueueItemRequest[] = [];
-        appointmentIds.forEach(a => {
-            const record = this.respondedTransactions.get(a);
-            if (record) queueItems.push(record.queueItem);
-            else throw new ArgumentError("No record of transaction in responder.", a);
-        });
-
-        for (const item of queueItems) {
-            // TODO:198: add back the contains and difference functions
-            if (this.mQueue.queueItems.findIndex(i => i.request.identifier.equals(item.identifier)) === -1) {
-                missingQueueItems.push(item);
-            }
-        }
-
+        const missingQueueItems = appointmentIdsStillPending
+            .map(a => this.respondedTransactions.get(a))
+            .map(a => {
+                if (!a) throw new ArgumentError("No record of appointment in responder.", a);
```

Comment:
perhaps give some meaningful names to the mapped variable, or at least two different ones to avoid having `a` with two different meanings in nearby rows.

Reply:
Updated